### PR TITLE
Docs

### LINKED ISSUE #57: docs: fix bullet lists by adding trailing whitespace
I noticed these aren't rendered as bullet lists since they are missing trailing whitespace (for some reason github's renderer doesn't care):

https://github.com/johannesjh/req2flatpak/blob/847406f3afabee316059f3a484466c70651541a8/docs/source/development.rst?plain=1#L29-L33

_Originally posted by @real-yfprojects in https://github.com/johannesjh/req2flatpak/issues/55#issuecomment-1512657372_

### PR DESCRIPTION
**Pull request by johannesjh**
* [x] adds empty lines between bulletpoints
* [x] updates doc-related dependencies
* [x] fixes bullet point rendering #57
* [x] make sure we publish docs whenever a tag is built

fixes #57

### CODE REVIEW

**Comment by johannesjh** [comment]
I added whitespace around bullet points, this did not fix the bulletpoint rendering, at least not on my local machine.

I updated doc-related dependencies.  this is useful on it's own. it did however require to bump the minimum python version to 3.8. the updated dependencies should fix the bulletpoint rendering, judging from what I read over on stackoverflow https://stackoverflow.com/q/67542699, but in this case, it did not fix the issue, at least not on my local machine.

I am pushing the changes to see what the ci-built docs look like. to my surprise, the bulletpoints are now rendered correctly, see https://johannesjh.github.io/req2flatpak/docs/introduction.html

**Comment by johannesjh** [comment]
last checkbox, I wanted to review and verify that we actually publish docs for tags that are being build. 

* regarding previous tags: in the [gh-pages branch](https://github.com/johannesjh/req2flatpak/tree/gh-pages), documentation for the tag `v0.1` exists, but versions.txt does not reference it. It may be that I accidentally deleted the reference in versions.txt when I manually cleaned it up a while a go.
  * this means I could try to manually add `v0.1` to versions.txt again

* regarding future tags: the ci script looks good to me: https://github.com/johannesjh/req2flatpak/blob/f898a19ef26036ff11b912085c6451732899bfb9/.github/workflows/ci.yml#L91

**Comment by real-yfprojects** [comment]
> I added whitespace around bullet points, this did not fix the bulletpoint rendering, at least not on my local machine.

Same for me however updating dependencies did the trick. Did you `--sync` your poetry environment afterwards?

> regarding future tags: the ci script looks good to me

I saw that you copied it from `sphinx_rtd_theme_github_versions` however it sometimes overrides `version.txt`: https://github.com/johannesjh/req2flatpak/commit/a0100826eeed8c7cb0f8028f94e08c681e5af40b 
I am currently trying to find the reason for that.

**Comment by real-yfprojects** [comment]
Ci output:
```
> Run poetry run sphinx_rtd_theme_github_versions .github/pages
fatal: Not a valid object name origin/gh-pages
WARNING:root:Cannot get gh-pages branch contents
Local directory: docs
Tags list: ['']
Sorted versions: ['docs']
```

Maybe running `git fetch origin gh-pages` fixes that.

**Comment by real-yfprojects** [comment]
I ticked the checkbox because I think this is fixed now.

File: pyproject.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ readme      = ["README.rst", "docs/source/changelog.rst"]
 req2flatpak = 'req2flatpak:main'
 
 [tool.poetry.dependencies]
-python    = "^3.7.2"
+python    = "^3.8"
```
Comment (real-yfprojects) [inline pyproject.toml]: Not sure we want to drop `3.7`. This is a very simple build script that should easily support most python versions. `py3.7` is still used in many distributions or on many installations afaik.

Updating `sphinx-rtd-theme` to `^1.2.0` already fixes the issue. No need to update `sphinx`.

Comment (johannesjh) [reply pyproject.toml]: I agree. I'll re-do the dependency updating, keeping compatibility with py3.7

Comment (johannesjh) [reply pyproject.toml]: done.

**Comment by johannesjh** [comment]
I downgraded sphinx to stay compatible with py3.7. 
=> I think we are all set, ready for merging.

**Review by real-yfprojects — APPROVED** [review]

**Pull request merged**